flexible_event_config.md

```diff
@@ -169,6 +170,15 @@ When the `event_report_window` for a spec completes, we will map its summary val
   "trigger_summary_bucket": [<bucket start>, <bucket end>]
 }
 ```
+### Trigger prioritization
+
+Given that triggering attribution can affect a source's state without producing a report, we will need a new algorithm for doing trigger prioritization. Here is a sketch of how it could work:
+
+1. Always maintain a sorted list of triggers, sorted in order of priority (descending) and trigger time (ascending)
```

Comment:
Always maintain a sorted list of triggers **that are not fully consumed in sent report** , sorted in order of priority (descending), **then** trigger time (ascending).

Reply:
What does "fully consumed" mean here?

Reply:
Trigger value was fully applied in the creation of a report whose report time is equal to or earlier than the trigger time (for "count" operator, the value is 1).

Reply:
(Current trigger time, so report time less than current window start).

Reply:
In the context of this PR, I don't understand how a trigger could be partially consumed. Not that this PR goes into the details of summary buckets, but in step 2.1 is the trigger's entire value not added to the spec's current value, resulting in N reports depending on the summary buckets?

Reply:
If the previous report window reached bucket 10 with a trigger whose value is 12, we have a value of 2 left to consume in this window. If the trigger's value was 10, it was fully applied.

Reply:
Why wouldn't the spec's value be 12 at the end of the previous window instead of 10?

Reply:
All of this may be a bit of implementation details. Yan is pointing out that we don't really need to store triggers whose value has already fully contributed to sent reports.

Reply:
Right, triggers are removed in step 2.3.

Reply:
@yanzhangnyc let me know if step 2.3 satisfies you. I updated the wording on the sorting aspect in the latest commit.

Reply:
> @yanzhangnyc let me know if step 2.3 satisfies you. I updated the wording on the sorting aspect in the latest commit.

I think step 2 and 3 are not "wrong" but some wording may cause misleading. Commented in []
For 2, Flush [Pass?] all of the speculative reports that are scheduled to be emitted in the current window [of given trigger data], and update the source's state [about the matched triggers?] based on all of the triggers that were successfully applied.  

For 3, Erase all of the triggers associated [I think we should emphasize the trigger has been fully consumed, not only associated as discussed before] with the current spec and window from the source's trigger list

Reply:
- Regarding [of the given trigger data] how about "of the given spec", since we are operating on a spec's window which can operate across many trigger data simultaneously?
- Regarding "about the matched triggers", I don't understand this comment or what it is clarifying, isn't the clause afterwards "based on all the triggers that were successfully applied" saying a similar thing?
- Regarding "fully consumed", can you elaborate on the concern? Why isn't deleting all triggers associated with the spec + window sufficient?

Reply:
For the original PR, I think we can publish as a general introduction as it is. Here is discussing to clarify some details. 
* In Flex Event API, the report window is affiliated with trigger data, and not an independent concept. When we mentioning do something at end of report window, there will be two understanding about the operation is applied to all triggers/reports or only to the triggers/reports to this trigger data that related to this window. 
* I believe you are indicating the field "attributed triggers" in the Source, right? The term "source state" sounds like a simple state like "deleted", "active", "expired". But it is OK to leave as it is. 
* Suppose a trigger have value of 12. The value of 8 has contributed to a report that have been sent so the trigger is "associated", but there are value of 4 left can be used to next report so it should not been "deleted".

Reply:
@yanzhangnyc I think you're describing an implementation, not an explainer-level algorithm description, particularly with the idea that part of a trigger's value is left over for another report in your third item.

Reply:
> Why isn't deleting all triggers associated with the spec + window sufficient?

Let's say we consumed 10 of a trigger's value of 12. If we erase this trigger, we have no way of knowing the priority of the remaining value of 2 as it applies to the procedure of generating reports in the next window.

Reply:
> > Why isn't deleting all triggers associated with the spec + window sufficient?
> 
> Let's say we consumed 10 of a trigger's value of 12. If we erase this trigger, we have no way of knowing the priority of the remaining value of 2 as it applies to the procedure of generating reports in the next window.

That is an implementation detail not relevant to the explainer-level description here and is covered by "update the source's state based on all of the triggers that were successfully applied."

Reply:
Ok, then if nothing else, this shows how hard it is for me to understand the current proposal.

Reply:
What does "update the source's state based on all of the triggers that were successfully applied" mean?